src/vm/diagnosticprotocolhelper.cpp

```diff
@@ -11,17 +11,57 @@
 #ifdef FEATURE_PERFTRACING
 #ifdef FEATURE_PAL
 
-static void WriteStatus(uint64_t result, IpcStream* pStream)
+void DiagnosticProtocolHelper::HandleIpcMessage(DiagnosticsIpc::IpcMessage& message, IpcStream* pStream)
 {
-    uint32_t nBytesWritten = 0;
-    bool fSuccess = pStream->Write(&result, sizeof(result), nBytesWritten);
-    if (fSuccess)
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+        PRECONDITION(pStream != nullptr);
+    }
+    CONTRACTL_END;
+
+    switch ((DumpCommandId)message.GetHeader().CommandId)
     {
-        fSuccess = pStream->Flush();
+    case DumpCommandId::GenerateCoreDump:
+        DiagnosticProtocolHelper::GenerateCoreDump(message, pStream);
+        break;
+
+    default:
+        STRESS_LOG1(LF_DIAGNOSTICS_PORT, LL_WARNING, "Received unknown request type (%d)\n", message.GetHeader().CommandSet);
+        DiagnosticsIpc::IpcMessage::SendErrorMessage(pStream, DiagnosticsIpc::DiagnosticServerErrorCode::UnknownCommand);
+        delete pStream;
+        break;
     }
 }
 
-void DiagnosticProtocolHelper::GenerateCoreDump(IpcStream* pStream)
+const GenerateCoreDumpCommandPayload* GenerateCoreDumpCommandPayload::TryParse(BYTE* lpBuffer, uint16_t& BufferSize)
```

Comment:
To avoid `DiagnosticsProtocolHelper.cpp` becomes another `debugger.cpp`, I advise we should move the individual payloads into separate files.

Reply:
I'd be ok with every CommandSet having its own file and perhaps that is what you meant, but isolating every payload in its own file feels like a principle that leads to very tiny files. I'd rather we not go that far... or at least someone would need to give me some rationale why bigger files are bad and numerous tiny files are superior. Its not a universally held belief : )

Reply:
I envisioned every `CommandSet` having its own file, so we don't end up with a `CommandId` switch case being in a different file from the functions that are dispatching the work for that command.  If I am splitting the Profiler and Dump commands into their own sets, that would mean having a `DumpDiagnosticProtocolHelper.h|cpp` and a `ProfilerDiagnosticProtocolHelper.h|cpp`.  It would be cleaner to maybe move all these protocol helpers inside their own directory, but I'm hesitant to deviate from the flat hierarchy the coreclr src seems to prefer (I'd be easily convinced that's the correct move though!).

Reply:
I don't think you need to create another directory for them